### ISSUE TITLE
New rule: disallow to cast to nullable type

### ISSUE THREAD

**Issue by t-kameyama** [detekt/detekt#3479]
noncompliant:
```kotlin
fun foo(a: Any?) {
    val x: String? = a as String?
    // If 'a' is not String, ClassCastException will be thrown.
}
```

compliant:
```kotlin
fun foo(a: Any?) {
    val x: String? = a as? String
}
```

**Comment by BraisGabin** [comment]
Why is this related with a nullable type? This has the same problem that your noncomoliant, right?

```kotlin
fun foo(a: Any) {
    val x: String = a as String
    // If 'a' is not String, ClassCastException will be thrown.
}
```

And with this we are flagging all the casts. They are dangerous but I think that we should not create a rule about it.

**Comment by t-kameyama** [comment]
There are cases where `as String?` is misused as safe casting (`as? String`), and so I want to detect that.

**Comment by BraisGabin** [comment]
I agree that there are some confusion between `as? String` and `as String?` but I don't know how to create a rule that flag those without a lot of false positves.
